lib/spack/spack/relocate.py

```diff
@@ -482,19 +511,25 @@ def replace(match):
         ndata = pat.sub(replace, data)
         if not len(ndata) == original_data_len:
             raise BinaryStringReplacementError(
-                path_name, original_data_len, len(ndata))
+                filename, original_data_len, len(ndata))
         f.write(ndata)
         f.truncate()
 
 
-def replace_prefix_nullterm(path_name, old_dir, new_dir):
-    """
-    Attempt to replace old install prefix with new install prefix
-    in binary files by replacing with null terminated string
-    that is the same length unless the old path is shorter
-    Used on linux to replace mach-o rpaths
-    """
+def _replace_prefix_nullterm(filename, old_dir, new_dir):
```

Comment:
@gartung I noticed that this function is not used at all in the code base. Is it a good candidate for removal or was it maintained on purpose for some reason?

Reply:
I use it to set rpaths in mach-o binaries when they are installed on a linux server.

Reply:
But I see that I dropped that case at some point in favor of just making making mach-o dependency entires use the full path.

Reply:
I'll remove it then. In case we need it later we can retrieve it from git history. Is that fine with you?

Reply:
OK. This an attempt to reproduce what conda does.